docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+Let's send the `hi` action to the local node and send as input parameter the test account name you created earlier:
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+The output of the command above shows on the first line that `hello::hi` action was executed with the input parameter `{"nm":"ama"}` and on the second line the output of the action itself `Name: ama`.
+
+```txt
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name: ama
+```
+
+## Smart Contract Source Files
+
+The smart contract C++ source files are:
+
+- hello.hpp
+- hello.cpp
+
+### The hello.hpp File
+
+The `hello.hpp` is the C++ header file and it contains the `hello` smart contract C++ class definition.
+
+```c++
+#include <eosio/eosio.hpp>
+using namespace eosio;
+
+CONTRACT hello : public contract {
+   public:
+      using contract::contract;
+
+      ACTION hi( name nm );
+};
+```
+
+A smart contract class definition must:
+
+- Be annotated by the [[eosio::contract]] attribute which tells the compiler it is a smart contract class; in the `hello.hpp` generated code the `CONTRACT` macro is used, which expands to `class [[eosio::contract]]` C++ code.
+- Be derived from the `contract` class which provides basic smart contract functionality.
+- Define at least a public action function, in this case it has only one, the `hi` action.
+- Define a public action wrapper for each action defined.
+
+You will learn more about actions later.
+
+### The hello.cpp File
+
+The `hello.cpp` is the C++ implementation file and it contains the `hello` smart contract C++ class implementation, that is, the implementation for every member function of the class.
+
+## Actions
+
+An action is a method, defined and implemented by a smart contract class, that can be called by an authorized account to perform some specific task or operation on the blockchain. Actions can be called by other contracts or by external accounts with the EOS Chain API.
+
+Actions can have parameters and return values and their responsibility is to execute the business logic of the contract. For example, an action in a token contract might transfer tokens from one account to another or update the balances of multiple accounts.
+
+Each action requires a specific level of authorization, which can be specified in the action's code. This ensures that only authorized accounts can execute specific actions.
+
+The `hello` smart contract class has only one action implemented by its `hi` public member function.
+
+```c++
+#include <hello.hpp>
+
+ACTION hello::hi( name nm ) {
+   /* fill in action body */
+   print_f("Name : %\n",nm);
+}
+```
+
+The function that implements a smart contract action must be annotated by the `[[eosio::action("action.name")]]` attribute. The `action.name` is optional and if not specified then the action is named by the function name that implements it.
+In the `hello.cpp` generated code the `ACTION` macro is used which expands to `[[eosio::action]] void` C++ code.
```

Comment:
This is really the basics tutorial, let's stick with just talking about the macros please. 
(Without actually calling them macros, which leads to more confusion for devs)

Just say that every action must be prefixed with ACTION or something. 
Same with CONTRACT above. It's good information for those that really wanna dig deep into the fact that the macros expand to the annotations, but it's not necessary education for the majority.

Reply:
I see your point, however later we explain the action name CAN have "." in it. And if the user tries he will fail with c++ compilation error. This section, including the next, allows the user to understand the limitations of using the ACTIONS macro and how he can actually avoid them limitations to name his actions anything he wants to and is allowed to.
if this is ok with you, I will remove/reword this to not mention macros and attributes.

Reply:
Sounds good.

We can also add a little "info box" which tells a user what a MACRO is. 

We can teach them c++ as we go, just don't want to assume they know it already or throw things at them they have to Google.

Reply:
I added the following:

[[info | C++ Macro]]
| In C++, a `macro` is a way to define a shorthand for a piece of code or a value. When the `macro` is used in the code, the preprocessor automatically replaces it with its corresponding definition before the code is compiled. This can help to make the code more readable and maintainable by reducing repetition and increasing abstraction.